### ISSUE TITLE
Accuracy is not correctly implemented

### ISSUE THREAD

**Issue by helinwang** [KarimJedda/dahoam2017#4]
https://github.com/KarimJedda/dahoam2017/blob/master/solver/captcha_model.py#L112

```python
correct_prediction = tf.equal(tf.argmax(logits,2), tf.argmax(labels,2))
```

`tf.equal` is element-wise:

```
>>> a = [[1, 1, 1], [1,1,1]]
>>> b = [[1,0,0], [0,0,0]]
>>> tf.equal(a, b).eval()
array([[ True, False, False],
       [False, False, False]], dtype=bool)
```

The code is true unless the result of `tf.equal(a, b).eval()` is `array([[False],
       [False]], dtype=bool)`, which is not the case.

**Comment by KarimJedda** [comment]
I know. That's why the next line is a reduce.

I might got the nomenclature wrong on that one.

helinwang <notifications@github.com> schrieb am Fr. 29. Sep. 2017 um 17:58:

**Comment by helinwang** [comment]
Oh, thanks for the explanation! I forgot about the code below.
